### ISSUE TITLE
Trying to decide whether to require_state_in_address

### ISSUE THREAD

**Issue by jjthames** [drhenner/ror_ecommerce#135]
I'm going back and forth on this and was hoping you could help. Ideally I would like to let people sign up and enter addresses in countries that don't use states/provinces (e.g. Luxembourg). However, it looks like states are used for some fairly important things:

For instance, in the address model I found:

``` ruby
  # hash of all the address attributes to be passed to a creditcard processor
  #
  # @param none
  # @ return [Hash] address attributes for a creditcard processor
  def cc_params
    { :name     => name,
      :address1 => address1,
      :address2 => address2,
      :city     => city,
      :state    => state.abbreviation,
      :country  => state.country_id == Country::USA_ID ? 'US' : 'CAN',
      :zip      => zip_code#,
      #:phone    => phone
    }
  end
```

I could rewrite the above to pull the country from Address#country, but then I would be left without a state. Is that going to make AuthNet reject transactions?

Also, charging in-state sales tax (obviously) requires knowing what US state you're shipping to.

ShippingZones could only be set up at the national level.

I took a look at Tonx (one of your example sites) and it looks like they have "fake" states set up for New Zealand. I only say they are "fake" because from what I can tell, New Zealand addresses only use street address+city+postal code.

So to sum up my questions:
1. Do you know if it will break anything to remove the state requirement?
2. Have you ever done a hybrid approach where the state is optional, or where the state is required only in certain countries?
